server/handlers.go

```diff
@@ -134,3 +134,81 @@ func (server HTTPServer) getRuleIDs(writer http.ResponseWriter, request *http.Re
 		return
 	}
 }
+
+// overviewEndpoint returns a map with an overview of number of clusters hit by rules
+func (server HTTPServer) overviewEndpoint(writer http.ResponseWriter, request *http.Request) {
+	authToken, err := server.GetAuthToken(request)
+	if err != nil {
+		handleServerError(writer, err)
+		return
+	}
+
+	userID := authToken.AccountNumber
+	orgID := authToken.Internal.OrgID
+
+	clustersHits := 0
+	var hitsByTotalRisk map[int]int = make(map[int]int)
+	var hitsByTags map[string]int = make(map[string]int)
+
+	clusters, err := server.readClusterIDsForOrgID(orgID)
+	if err != nil {
+		handleServerError(writer, err)
+		return
+	}
+
+	for _, clusterID := range clusters {
+		aggregatorResponse, successful := server.readAggregatorReportForClusterID(orgID, clusterID, userID, writer)
+		if !successful {
+			log.Info().Msgf("Aggregator doesn't have reports for cluster ID %s", clusterID)
+			continue
+		}
+
+		if aggregatorResponse.Meta.Count == 0 {
+			continue
+		}
+		clustersHits = clustersHits + 1
+
+		var cacheTotalRisk map[int]bool = make(map[int]bool)
+		var cacheTags map[string]bool = make(map[string]bool)
+		for _, report := range aggregatorResponse.Report {
+			ruleID := report.Module
+			errorKey := report.ErrorKey
+
+			ruleWithContent, err := content.GetRuleWithErrorKeyContent(ruleID, errorKey)
+			if err != nil {
+				handleServerError(writer, err)
+				return
+			}
+
+			if val, found := cacheTotalRisk[ruleWithContent.TotalRisk]; !found || val {
```

Comment:
What's the point of this? It's always true, isn't it?

Reply:
When you iterate for a given report in a given cluster, at least the first hit won't be already in this map. Both `cache*` vars will be re-created when the first `for` iterate to the next cluster

Reply:
And what's the point? You don't use this cache anywhere and your `if` is hit every time. Or am I missing something?

Reply:
The cache is used right here just to control when the `if` block is executed or not. Let me put an example: for a given cluster, you have a report with 3 rule hits. 1st and 3rd have the same "total risk" value, and 2nd a different one.

When you iterate over the rules in the report, first time you will check the `cacheTotalRisk`, you won't find the key and execute the `if` block, that sets the value in the `cacheTotalRisk` map and increase de real counter.

For the 2nd rule, the same will apply because the "total risk" is different.

For the 3rd rule, you will check the `cacheTotalRisk` and you will find a `true` value, so you don't enter inside the `if` block and avoid counting the same risk 2 times for this cluster